server/connector.go

```diff
@@ -342,7 +383,10 @@ func (c *Connector) readUserInfo(bufferedReader *bufio.Reader, clientAddr net.Ad
 func (c *Connector) findAndConnectBackend(ctx context.Context, frontendConn net.Conn,
 	clientAddr net.Addr, preReadContent io.Reader, serverAddress string, playerInfo *PlayerInfo, nextState mcproto.State) {
 
-	backendHostPort, resolvedHost, waker := Routes.FindBackendForServerAddress(ctx, serverAddress)
+	// Should always be the last defer statement
```

Comment:
Now that there's three defer's in the same function I'm thinking it would be cleaner to introduce a new 'cleanupBackendConnection` type method that becomes the target of a single defer. One of the things refactored into that function's args would be an indication of whether down scaling should begin.